File: Backend/src/models/note.js
//Steps
//01. Define Schema => Note : id , userID , title ,content ,dateadded
//02. create model => <model name> <schema> Note

const mongoose = require('mongoose');

const noteSchema = mongoose.Schema({
    id:{
        type: String,
        unique : true,
        required:true,
    },
    userID:{
        type:String,
        required:true,
    },
    content:{
        type:String,
    },
    dateAdded:{
        type:Date,
        default:Date.now,
    },
    title:{
        type:String,
        required:true
    }
});


module.exports = mongoose.model("Note",noteSchema);

File: flutter_note_app/lib/models/note.dart
class Note {
  String? id;
  String? userID;
  String? title;
  String? content;
  DateTime? dateadded;

  Note({this.id, this.content, this.dateadded, this.title, this.userID});

  factory Note.fromMap(Map<String, dynamic> map) {
    return Note(
      id: map["id"],
      userID: map["userID"],
      content: map["content"],
      dateadded: DateTime.tryParse(map["dateAdded"]),
      title: map["title"],
    );
  }

  Map<String, dynamic> toMap() {
    return {
      "id":id,
      "userID":userID,
      "title":title,
      "content":content,
      "dateAdded": dateadded!.toIso8601String(),
    };
  }
}
